lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -459,6 +459,11 @@ def __bool__(self):
     def __copy__(self):
         return KymoTrackGroup(copy(self._src))
 
+    def __add__(self, other):
+        new_group = copy(self)
+        new_group.extend(other)
```

Comment:
Probably, I'm missing something, but why do you make a shallow copy of the `KymoTrackGroup`? As I understand, the method `extend()` appends kymos to the list `self._src`. So after adding there are two different `KymoTrackGroup`s pointing to the same list of kymos, with the original `KymoTrackGroup` "silently" beeing changed. I would expect when performing something like `ktg3 = ktg1 + ktg2` that `ktg1` would stay unchanged.

Reply:
Yes, mystery solved. I originally stumbled because I missed the method `KymoTrackGroup.__copy__()`, which returns a copy of the list.